pages/videos.vue

```diff
@@ -31,17 +31,31 @@
     components: {
       VideoTable
     },
+    mounted() {
+      if (this.routeHasSearchQuery()) {
+        this.$refs.desktopTable.setSearchQuery(this.$route.query.q)
```

Comment:
Calling a child component's method from the parent feels like it could lead to spaghetti.  Let's avoid that, unless there's a reason why other methods don't work as well.

Reply:
Here's a better way to handle this.

Have `search` defined up here, and pass it down as a prop.  Then pass down as a prop a method that can change search.  Call that method when you need to change it in the component.

Data Down, Actions Up.